app/components/order-inline-textarea-input.js

```diff
@@ -1,60 +1,72 @@
-import Ember from 'ember';
```

Comment:
Although it's a great start having it take a callback as argument and removing the API call. It's still only usable for orders.

You could just have a generic `inline-textarea` component that sends the updated text when it changes : )
Building the `orderParams` is not the component's responsibility.

Imagine I want the exact same textinput for another property of another model, we don't want to be forced to re-write the whole component.

Analogy: the component is a car, and the controller is the driver. The car doesn't decide which route to take, it's just a tool which the driver can use to achieve his/her goal. It has a wheel, pedals, and you can use it in different ways.

Reply:
That make sense. I have renamed it and also pulled out Business logic from the component.